From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Gofman <pgofman@codeweavers.com>
Date: Mon, 9 Jun 2025 17:18:19 -0600
Subject: [PATCH] msi: HACK: Report older Windows version for Halo MCC redists.

CW-Bug-Id: #25520
Valve-commit: faf0f71be93cb0b53a19c9307c7c640d5f11cbaf
Picked: clean
Class: valve
Evidence: docs/evidence/2026-09-28-wine-proton-rebase.md
Offered-upstream: no
---
 dlls/msi/package.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/dlls/msi/package.c b/dlls/msi/package.c
index d20e87f..d0b546c 100644
--- a/dlls/msi/package.c
+++ b/dlls/msi/package.c
@@ -622,6 +622,25 @@ done:
     free(version);
 }
 
+static void fixup_winver(DWORD *verval)
+{
+    static int cached = -1;
+
+    if (cached == -1)
+    {
+        const char *s;
+
+        cached = (s = getenv("STEAM_COMPAT_APP_ID")) &&
+                    (
+                        !strcmp(s, "976730")
+                    );
+        if (cached)
+            ERR("HACK: setting winver 502.\n");
+    }
+    if (!cached) return;
+    if (*verval > 502) *verval = 502;
+}
+
 static VOID set_installer_properties(MSIPACKAGE *package)
 {
     WCHAR *ptr;
@@ -749,6 +768,7 @@ static VOID set_installer_properties(MSIPACKAGE *package)
         verval = 603;
         OSVersion.dwBuildNumber = 9600;
     }
+    fixup_winver(&verval);
     len = swprintf( verstr, ARRAY_SIZE(verstr), L"%u", verval );
     switch (OSVersion.dwPlatformId)
     {
